internal/jasmine_node_test/jasmine_runner.js

```diff
@@ -11,45 +12,113 @@ const UTF8 = {
 const BAZEL_EXIT_TESTS_FAILED = 3;
 const BAZEL_EXIT_NO_TESTS_FOUND = 4;
 
+// ibazel will write this string after a successful build
+// We don't want to re-trigger tests if the compilation fails, so
+// we should only listen for this event.
+const IBAZEL_NOTIFY_BUILD_SUCCESS = 'IBAZEL_BUILD_COMPLETED SUCCESS';
+const IBAZEL_NOTIFY_CHANGES = 'IBAZEL_NOTIFY_CHANGES';
+
 // Set the StackTraceLimit to infinity. This will make stack capturing slower, but more useful.
 // Since we are running tests having proper stack traces is very useful and should be always set to
 // the maximum (See: https://nodejs.org/api/errors.html#errors_error_stacktracelimit)
 Error.stackTraceLimit = Infinity;
 
-function main(args) {
-  if (!args.length) {
-    throw new Error('Spec file manifest expected argument missing');
-  }
-  const manifest = require.resolve(args[0]);
-  // Remove the manifest, some tested code may process the argv.
-  process.argv.splice(2, 1)[0];
+function watchMode(manifestLocation) {
+  // ibazel communicates with us via stdin
+  const rl = createInterface({input: process.stdin, terminal: false});
+  rl.on('line', chunk => {
+    if (chunk === IBAZEL_NOTIFY_BUILD_SUCCESS) {
+      // TODO: we should try keep the same instance of JasmineRunner if it's expensive to new up
+      // TODO: stop previous runs if we get multiple calls of this in quick succession (debounce)
+      const jrunner = createRunner(true);
+      addFiles(manifestLocation, jrunner);
+      jrunner.execute();
+    }
+  });
+  rl.on('close', () => {
+    // Give ibazel 5s to kill our process, otherwise do it ourselves
+    setTimeout(() => {
+      console.error('ibazel failed to stop jasmine after 5s; probably a bug');
+      process.exit(1);
+    }, 5000);
+  });
+}
+
+function deleteFromCache(fileName) {
+  // jasmine uses `require(fileName)` to read the files into the program
+  // so we have to delete them from the cache to ensure they read again on subsequent runs
+  // when in ibazel mode
+  delete require.cache[require.resolve(fileName)];
+}
+
+function addFiles(manifestLocation, jrunner){
+  deleteFromCache(manifestLocation)
+
+  const manifest = require.resolve(manifestLocation);
+  const file = fs.readFileSync(manifest, UTF8);
```

Comment:
seems like an extra read? what do you do with this variable?

Reply:
whops, was debugging with that, fixed